Realm/Realm.Fody/ModuleWeaver.cs

```diff
@@ -497,6 +497,10 @@ private WeaveResult WeaveProperty(PropertyDefinition prop, TypeDefinition type,
                               new GenericInstanceMethod(_references.RealmObject_GetListValue) { GenericArguments = { elementType } },
                               concreteListConstructor);
         }
+        else if (prop.PropertyType.GetElementType().FullName == "System.Collections.Generic.List`1")
+        {
+            return WeaveResult.Error($"{type.Name}.{prop.Name} is a 'System.Collections.Generic.List`1' but only its interface is supported by Realm. Did you mean IList?");
```

Comment:
We can reword this error message to be a bit more precise and actionable. I'm thinking of something like:

```
$"{type.Name}.{prop.Name} is declared as List<{*prop.genericargument*}> which is not the correct way to declare to-many relationships in Realm. If you want to persist the collection, use the interface IList<{*prop.genericargument*}, otherwise annotate the property with the [Ignored] attribute."
```

It's a bit more verbose, but it achieves a few things:
1. It makes `System.Collections.Generic.List`1` more readable.
2. Makes the `only its interface is supported by Realm` a bit more precise - `List<T>` implements multiple interfaces and while the most obvious one is `IList<T>`, it's imprecise to call it "its interface".
3. It tells people what to do in case they do want to have a `List<T>` property that is not persisted in Realm.

Reply:
What you're saying makes totally sense. Good points.
I'll apply this at the end of the skunworks week.